### DISCUSSION TITLE
Updating table properties

### DISCUSSION THREAD

**Discussion by capstoneanalytics** [TabularEditor/TabularEditor3#1219] [🙋♀️TE3 Q&A] [answered]
I have seen some conflicting information around the internet and maybe im confusing them myself but can you update table/partition properties like M expressions, Table descriptions using TE 3 ? I thought that I had achieved that at some point using PBIP files but doesnt seem to work now. This article does say you can edit table properties using Desktop

https://docs.tabulareditor.com/te3/views/properties-view.html

**Comment by otykier** [comment, marked as answer]
If you're working directly against Power BI Desktop, then no, [changes to M expressions or tables/partitions are not supported](https://learn.microsoft.com/en-us/power-bi/transform-model/desktop-external-tools#supported-write-operations).

If you're working against PBIP files, then it's possible, but [this requires a TE3 Business or Enterprise Edition license (TE3 Desktop Edition can only be used when working directly against Power BI Desktop)](https://docs.tabulareditor.com/te3/editions.html).

If you do not wish to upgrade your TE3 license, there is always the option to use TE2 to make changes to the Model.bim file in the PBIP folder structure.

**Comment by capstoneanalytics** [comment]
Thanks Daniel. I do have an enterprise license and I could make changes previously but now I cannot. I tried it again this morning in a new file and changed the description of a table in TE3 but it just doesnt save in the pbip file. Am I missing a step or a configuration here ?

**Reply by mlonsk** [reply]
Hi @capstoneanalytics 
Your license does not matter if you are using Tabular Editor as an External Tool from Power BI. 
I.e. If you open a .pbip file in Power BI Desktop and open Tabular Editor through the External Tool Ribbon will Power BI still apply the limitations of working against a Power BI file by e.g. not allowing changes to tables. See : [Power BI Desktop Limitations](https://docs.tabulareditor.com/te3/desktop-limitations.html?tabs=postjune2023)

What you need to do is to open Tabular Editor separately and then open the Model.bim file inside the2 .pbip folder structure, make your changes and save them in TE. When you then open your .pbip folder in Power BI desktop the changes will be reflected. Though you may still need to refresh the table, this will not remove the description.

**Reply by capstoneanalytics** [reply]
ah that was it, I knew it something very simple I was missing. It works like a charm :). Thank you.
